### ISSUE TITLE
X Error of failed request:  BadValue (integer parameter out of range for operation)

### ISSUE THREAD

**Issue by jose1711** [i3/i3#3586]
## I'm submitting a…
<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior
Running `xclock` from an already opened `xterm` window fails with the following error message:
```
X Error of failed request:  BadValue (integer parameter out of range for operation)
  Major opcode of failed request:  12 (X_ConfigureWindow)
  Value in failed request:  0x0
  Serial number of failed request:  86
  Current serial number in output stream:  8
```

## Expected Behavior
`xclock` windows should be opened on a configured workspace.

## Reproduction Instructions
* configure `i3` to open `xclock` windows on (currently) not existing workspace
```
$ grep XClock .config/i3/config 
for_window [class="^XClock$"] move window to workspace 8
```
* reload config
```
i3 reload
```
* make sure you're *not* switched to workspace 8
* run `xclock` from `xterm` window
```
xclock
# receive the error
```

## Environment
Output of `i3 --moreversion 2>&-`:
```
Binary i3 version:  4.16 (2018-11-04) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.16 (2018-11-04) (pid 14582) abort…)
Loaded i3 config: /home/jose/.config/i3/config (Last modified: So 12. január 2019, 14:58:20 CET, 608 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3-with-/i3
```

config: auto-generated by `i3-confg-wizard` + single custom line added
```
# This file has been auto-generated by i3-config-wizard(1).
# It will not be overwritten, so edit it as you like.
#
...
...

# Start i3bar to display a workspace bar (plus the system information i3status
# finds out, if available)
bar {
        status_command i3status
}

for_window [class="^XClock$"] move window to workspace 8
```
Relevant lines from debug log: https://logs.i3wm.org/logs/5706933096939520.bz2

```
- Linux Distribution & Version: Arch Linux 64bit (up-to-date)
- Nvidia 750Ti with 3 monitors, i3 installed from repository
- Are you using a compositor (e.g., xcompmgr or compton): nope
```

**Comment by orestisf1993** [comment]
I can confirm that this happens but I don't know where the problem is.

Can you please also open an issue in [xclock](https://gitlab.freedesktop.org/xorg/app/xclock)?

**Comment by Airblader** [comment]
Running it under xtrace might be useful to see what values exactly xclock is sending here. I sort of assume this to be more of an xclock issue as well, but we'll see.

**Comment by orestisf1993** [comment]
[out.txt](https://github.com/i3/i3/files/2752282/out.txt)

**Comment by Airblader** [comment]
> 0.141 000:<:005a: 20: Request(12): ConfigureWindow window=0x00e0000b values={width=0 height=0} 

I assume that's the issue, since 0x0 dimensions aren't allowed. However, there is a

> 0.141 000:>:0059: Event (generated) ConfigureNotify(22) event=0x00e0000a window=0x00e0000a above-sibling=None(0x00000000) x=0 y=0 width=0 height=0 border-width=1 override-redirect=f

so maybe the fake notification sent by i3 isn't correct.

**Comment by jose1711** [comment]
I am seeing the same symptoms with `vncviewer` (tightvnc) and `xterm` so not sure if it makes sense to file a bugreport for `xclock.

**Comment by orestisf1993** [comment]
Yes, the con's rect is set here: https://github.com/i3/i3/blob/5aa0459be0a8471226dabda620c028c149966b29/src/manage.c#L589 but a fake_absolute_configure_notify with width=height=0 is sent because of https://github.com/i3/i3/blob/5aa0459be0a8471226dabda620c028c149966b29/src/manage.c#L568-L569.
I think the problem is explained in this comment: https://github.com/i3/i3/blob/5aa0459be0a8471226dabda620c028c149966b29/src/manage.c#L575-L581

So, the `tree_render` inside `run_assignments` doesn't correctly set `nc`'s rect because its workspace is invisible but a configurenotify is still sent.

**Comment by jose1711** [comment]
@orestisf1993 thank you for this workaround, it's fine for the time being
